modules/packages/BLAS.chpl

```diff
@@ -19,103 +19,130 @@
-the need to wrap these with ``c_ptrTo``). As with BLAS calls in C, 
-the user is responsible for passing in array dimensions etc. Furthermore, 
-different array element types require using different functions.
-
-We refer the user to the existing CBLAS documentation for the usage
-of these functions.
 
 Chapel Level 3 BLAS API
 -----------------------
 
-For convenience, we provide wrappers around the Level 3 (matrix-matrix)
-BLAS functions. These determine the appropriate functions to call, based
-on the array element types, as well as the array dimensions. The other
-functionality is identical to the corresponding BLAS functions.
+For convenience, this module provides wrappers around the Level 3
+(matrix-matrix) BLAS functions. These determine the appropriate functions to
+call, based on the array element types, as well as the array dimensions. The
+other functionality is identical to the corresponding BLAS functions.
 
 The names of these routines are identical to the corresponding BLAS functions,
-except that the type prefix is dropped. For instance, ``gemm`` is the 
+except that the type prefix is dropped. For instance, ``gemm`` is the
 wrapper for the ``[sdcz]gemm`` routines.
 
 .. note::
  Chapel determines the dimensions of the matrices from the arrays that are
- passed in. However, if one is passing in a sub-array such that the array 
- elements are not contiguously stored in memory, then the user needs to 
+ passed in. However, if one is passing in a sub-array such that the array
+ elements are not contiguously stored in memory, then the user needs to
  pass in the leading dimension (```lda``` etc) to the array, just as they
  would in C. These default to 0, in which case Chapel determines the appropriate
  values based on the array passed in.
 
 
 */
-
-
 module BLAS {
 
+  use C_BLAS;
+
   use SysCTypes;
   require "cblas.h";
 
 
   /* Define row or column order */
-  enum Order {Row=101:c_int, Col=102};
+  enum Order {Row=101 : c_int, Col};
 
   /* Operation of matrix : none, transpose, or adjoint */
-  enum Op {N=111:c_int, T=112, H=113}; // NoTranspose, Transpose, Adjoint
+  enum Op {N=111 : c_int, T, H}; // NoTranspose, Transpose, Adjoint
 
   /* Storage for symmetric matrices */
-  enum Uplo {Upper=121:c_int, Lower=122};
+  enum Uplo {Upper=121 : c_int, Lower};
 
   /* Assume a unit or non-unit diagonal */
-  enum Diag {NonUnit=131:c_int, Unit=132};
+  enum Diag {NonUnit=131 : c_int, Unit};
 
   /* Operate on the left or right side */
-  enum Side {Left=141:c_int, Right=142};
+  enum Side {Left=141 : c_int, Right};
 
   /* Level 3 BLAS */
 
-  /* Wrapper for the GEMM routines */
+  /*
+    Wrapper for the `GEMM routines <http://www.netlib.org/lapack/explore-html/db/def/group__complex__blas__level3_gac4e11e8e8a4b0c802d2229b08da341f6.html#gac4e11e8e8a4b0c802d2229b08da341f6>`_
+
+    Performs the matrix-matrix operation::
+
+      C = alpha * op(A) * op(B) + beta * C
+
+    where ``A`` is a triangular matrix.
```

Comment:
Nope, A is an arbitrary matrix. I think we just delete this sentence, otherwise we should add that A,B and C are general matrices and alpha and beta are scalars.